optics-core/src/Optics/Label.hs

```diff
@@ -87,9 +106,196 @@
 --
 -- >>> peter & set #pets []
 -- Human {humanName = "Peter", humanAge = 13, humanPets = []}
+
+-- $guide
+--
+-- == The problem
+--
+-- Standard Haskell records are a common source of frustration amongst seasoned
+-- Haskell programmers. Their main issues are:
+--
+-- 1) Pollution of global namespace as every field accessor is also a top-level
+--    function.
+--
+-- 2) Inability to define multiple data types sharing field names in the same
+--    module.
+--
+-- 3) Clunky update syntax, especially when nested fields get involved.
+--
+-- Over the years multiple language extensions were proposed and implemented to
+-- alleviate these issues. We're quite close to having a reasonable solution
+-- with the following trifecta:
+--
+-- - <https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#extension-DuplicateRecordFields DuplicateRecordFields> - introduced in GHC 8.0.1, addresses (2)
+--
+-- - <https://github.com/ghc-proposals/ghc-proposals/pull/160 NoFieldSelectors> - accepted GHC proposal, addresses (1)
+--
+-- - <https://github.com/ghc-proposals/ghc-proposals/pull/282 RecordDotSyntax> - accepted GHC proposal, addresses (3)
+--
+-- It needs to be noted however that both NoFieldSelectors and RecordDotSyntax
+-- are not yet implemented, with the latter depending on adding setField to
+-- HasField (<https://gitlab.haskell.org/ghc/ghc/issues/16232 ghc/16232>), also not
+-- yet implemented.
```

Comment:
Or at least not yet merged :wink: https://gitlab.haskell.org/ghc/ghc/-/merge_requests/3257 although that is mostly my fault for not pushing harder on getting feedback and help with the performance issues.

Reply:
Thanks, I changed it to "not yet merged".